### PR TITLE
Changes to Google sheets plugin's request and response structures

### PR DESCRIPTION
**Pull request by nidhi-nair**
Ref: #1729 

A few assumptions while designing the structure:
1. We're prioritizing database like operations with this integration
2. This plugin is meant to be used with table headers
3. For all intents and purposes, users as well as developers will refer to row indices by the relative index wrt the table header.
4. Data inputs are maps, and hence need not maintain ordering while inserting or updating
5. Empty rows are valid values
6. Empty columns are valid values
7. Missing values are mapped to empty cells
8. Missing headers are filled up within Appsmith as "Column-1", "Column-2", etc, depending on column number. This will be used only within Appsmith and will not reflect on the Sheet itself
9. Duplicate headers are renamed as "Header", "Header_1", "Header_2". This will be used only within Appsmith and will not reflect on the Sheet itself
10. Malformed keys are ignored if they are not found in the structure of the given table header index

**Known bugs:**
1. Fields that should not be visible are visible because of limited conditionals (row limit)
2. Range and row offset need to be separate fields for clarity, This cannot be supported right now because again there is no way to make one of them disappear while the other exists. (complicated conditional)
3. Body cannot show individual requests expected templates. This will be done via documentation for now
4. How should these methods be ordered?

### CODE REVIEW

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java [hunk 1]
```diff
@@ -2,38 +2,201 @@
...
+                .build();
+        final GetValuesMethod getValuesMethod = new GetValuesMethod(this.objectMapper);
+
+        List<RowObject> rowObjectListFromBody = null;
+        try {
+            rowObjectListFromBody = this.getRowObjectListFromBody(this.objectMapper.readTree(body));
+        } catch (JsonProcessingException e) {
+            // Should never enter here
         }
+        assert rowObjectListFromBody != null;
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java:80]: I'm curious; what happens to the flow of the HTTP request when an assert like this fails? What response do we end up sending to the client?

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java [hunk 2]
```diff
@@ -2,38 +2,201 @@
...
+                        return Mono.error(new AppsmithPluginException(
+                                AppsmithPluginError.PLUGIN_ERROR,
+                                "Could not map request back to existing data"));
+                    }
+                    String jsonBody = new String(responseBody);
+                    JsonNode jsonNodeBody = null;
+                    try {
+                        jsonNodeBody = objectMapper.readTree(jsonBody);
+                    } catch (IOException e) {
+                        Mono.error(new AppsmithPluginException(
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java:115]: Should we be returning this `Mono.error` ?

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java [hunk 3]
```diff
@@ -2,38 +2,201 @@
...
+                    // This is the object with the original values in the referred row
+                    final JsonNode jsonNode = getValuesMethod
+                            .transformResponse(jsonNodeBody, methodConfig);
+
+                    // This is the rowObject for original values
+                    final RowObject returnedRowObject = this.getRowObjectListFromBody(jsonNode).get(0);
+
+                    // We replace these original values with new ones
+                    finalRowObjectListFromBody.forEach(rowObject -> {
+                        final LinkedHashMap<String, String> valueMap = new LinkedHashMap<>(returnedRowObject.getValueMap());
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java:131]:
```suggestion
                        final Map<String, String> valueMap = new LinkedHashMap<>(returnedRowObject.getValueMap());
```

We should typically use interfaces to define the variable. The actual implementation will be determined by the initialization.

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java (hunk 1 above)
Comment (nidhi-nair) [reply app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/AppendMethod.java:80]: It results in an Assertion error, but these cases are only expected to happen if for some reason the API response structure changes, which it never should.

**Review by trishaanand — APPROVED** [review]

**Pull request merged**